Adding 'single_word_only' option to obfuscate processor

Utkarsh-Aga opened the pull request:
### Description
Adding `single_word_only` configuration option to obfuscate processor. When set to `true` a word boundary `\b` is added to the pattern, due to which obfuscation would be applied only to words that are standalone in the input text. By default, it is `false`, meaning obfuscation patterns are applied to all occurrences.
 
### Issues Resolved
Resolves #4340 
 
### Check List
- [X] New functionality includes testing.
- [X] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

dlvenable commented on data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/ObfuscationProcessor.java:
```diff
@@ -42,6 +42,7 @@ public class ObfuscationProcessor extends AbstractProcessor<Record<Event>, Recor
 
     private final String source;
     private final String target;
+    private final boolean SingleWordOnly;
```
Minor: Please use the Java convention of starting field names with a lowercase character.

Thus:

```
    private final boolean singleWordOnly;
```

Utkarsh-Aga replied:
Updated the variable name.

dlvenable reviewed (changes requested):
@Utkarsh-Aga , Thank you for this contribution. I only have one small comment.

Utkarsh-Aga commented:
Thanks a lot @dlvenable for the review, I have addressed that comment also added a additional unit test `testProcessorWithMultiplePatternsWithSingleWordOnly`.

dlvenable reviewed (approved):
Thank you @Utkarsh-Aga !

chenqi0805 reviewed (approved):
@Utkarsh-Aga Thanks for the change!
